### ISSUE TITLE
[CI] RegressionIT testWithOnlyTrainingRowsAndTrainingPercentIsFifty failure

### ISSUE THREAD

**Issue by martijnvg** [elastic/elasticsearch#50313]
```
2> java.lang.AssertionError: 
    Expected: <1>
         but: was <0>
        at __randomizedtesting.SeedInfo.seed([E1492A6333408A56:5D6C33F3033F5FA3]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ml.integration.RegressionIT.assertModelStatePersisted(RegressionIT.java:285)
        at org.elasticsearch.xpack.ml.integration.RegressionIT.testWithOnlyTrainingRowsAndTrainingPercentIsFifty(RegressionIT.java:169)
```

Reproduce line doesn't reproduce for me locally:

```
./gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner' --tests "org.elasticsearch.xpack.ml.integration.RegressionIT.testWithOnlyTrainingRowsAndTrainingPercentIsFifty" -Dtests.seed=E1492A6333408A56 -Dtests.security.manager=true -Dtests.locale=lt-LT -Dtests.timezone=Canada/Eastern -Dcompiler.java=12 -Druntime.java=8
HomeTestsRegressionIT » testWithOnlyTrainingRowsAndTrainingPercentIsFifty
```

Build log: https://gradle-enterprise.elastic.co/s/mfwgongqmuwl6

Failure is related to this: https://github.com/elastic/elasticsearch/issues/47945#issuecomment-544414079

That issue got closed in favour of another issue, which got closed with a fix. However this failure started appear again as can be seen here: 
https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now%2FM,mode:quick,to:now%2FM))&_a=(columns:!(_source),filters:!(),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'test:testWithOnlyTrainingRowsAndTrainingPercentIsFifty'),sort:!(time,desc))

**Comment by mayya-sharipova** [comment]
This failed on 7.x today with a couple of other tests in `RegressionIT` and `ClassificationIT`.

Log: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+release-tests/986/console
Build Scans: https://gradle-enterprise.elastic.co/s/doq7yy73r32cy

```
org.elasticsearch.xpack.ml.integration.RegressionIT > testWithOnlyTrainingRowsAndTrainingPercentIsFifty FAILED |  
-- | --
  | java.lang.AssertionError: |  
  | Expected: a value greater than <0> |  
  | but: <0> was equal to <0> |  
  | at __randomizedtesting.SeedInfo.seed([EAEE72213F9A9746:56CB6BB10FE542B3]:0) |  
  | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  | at org.junit.Assert.assertThat(Assert.java:956) |  
  | at org.junit.Assert.assertThat(Assert.java:923)
```

**Comment by benwtrent** [comment]
```
RegressionIT#testSingleNumericFeatureAndMixedTrainingAndNonTrainingRows
```
Failed as well. Looking at the code:
https://github.com/elastic/elasticsearch/blob/b84e8db7b5a3da90444afec830e210920e0c164f/x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/RegressionIT.java#L89


That is exceptionally strange. The `is_training` is true, but the dest doc did not include the dependent variable. 

`is_training` is determined by if the dependent variable is present or not. Recent changes have modified what "missing" means (changing `""` to `\0`). Something might have broken there?

**Comment by mark-vieira** [comment]
We're getting a lot of failures from `RegressionIT` and `ClassificationIT` probably worth a mute?

**Comment by droberts195** [comment]
I think https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+release-tests/986/console failed because of a mismatch between the C++ and Java code.  Because it's a "release-tests" build it requires a _non_-snapshot C++ build, and I think the one that existed at the time this CI job ran was too old.

Let's just monitor whether this continues to fail in the same way over the next day or two.  If it mysteriously stops failing then it was due to the _non_-snapshot C++ being out of date.  If it carries on, or is seen in a build that isn't "release-tests" then it's a genuine bug.

**Comment by mark-vieira** [comment]
I think the failures I'm referring to are #53188, the seem to be different than this issue.

**Comment by droberts195** [comment]
> We're getting a lot of failures from `RegressionIT` and `ClassificationIT` probably worth a mute?

Different tests in these suites are failing for different reasons.  #53188 is still failing on 7.x because it was only muted on master.  That one is worth muting.

**Comment by ywangd** [comment]
Had a few more failures for `7.7` release-tests today that are related to this. In addtion to `RegressionIT`, they also failed for the following classes (50+ failed methods in total):
* `ClassificationIT`
* `DatafeedJobsRestIT`
* `MlJobIT`
* `InferenceIngestIT`

The `RegressionIT.testWithOnlyTrainingRowsAndTrainingPercentIsFifty` however failed with **timeout** error (`Test abandoned because suite timeout was reached`) which is reproducible locally with:
```
./gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner' --tests "org.elasticsearch.xpack.ml.integration.RegressionIT.testWithOnlyTrainingRowsAndTrainingPercentIsFifty" \
  -Dtests.seed=349DA09002B31208 \
  -Dtests.security.manager=true \
  -Dbuild.snapshot=false \
  -Dtests.jvm.argline="-Dbuild.snapshot=false" \
  -Dtests.locale=ro \
  -Dtests.timezone=Antarctica/Rothera \
  -Dcompiler.java=13 -Dlicense.key=x-pack/license-tools/src/test/resources/public.key
```

Other errors are not reproducible. A sample error message is as the follows:
```
:x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner » org.elasticsearch.xpack.ml.integration.DatafeedJobsRestIT » testLookbackOnlyGivenAggregationsWithHistogramAndRollupIndex (46.127s)
3 active tasks found:
xpack/ml/data_frame/analytics[c]                 dubc7glwS8KCmNntTfQWRQ:14794 cluster:30                   persistent 1585728316208 08:05:16 37.9m       127.0.0.1 integTest-2 data_frame_analytics-regression_only_training_data_and_training_percent_is_50
cluster:admin/xpack/ml/data_frame/analytics/stop dubc7glwS8KCmNntTfQWRQ:15363 -                            transport  1585728349153 08:05:49 37.4m       127.0.0.1 integTest-2 
cluster:admin/xpack/ml/data_frame/analytics/stop b7l42R8jQO-LqXcWf37MyQ:8597  -                            transport  1585728349153 08:05:49 37.4m       127.0.0.1 integTest-0 
 expected:<0> but was:<3>
```

Note that I added `-Dlicense.key=x-pack/license-tools/src/test/resources/public.key` at the end of the reproducation line, otherwise it won't run.

Build scans:
* https://gradle-enterprise.elastic.co/s/d3hukbf64r72e
* https://gradle-enterprise.elastic.co/s/6dyjgcv6kyqi4
* https://gradle-enterprise.elastic.co/s/q3vxfhecfogne

**Comment by droberts195** [comment]
The most recent problem in the 7.7 release tests was caused by another Java/C++ mismatch.  This is resolved now and the 7.7 release tests after the 3 that failed succeeded: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.7+release-tests/29/

It didn't fail in between 2 and 25 days ago, so I think all the various problems are now resolved and I will close this now.
